artifactory/utils/container/buildinfo.go

```diff
@@ -119,6 +133,135 @@ func setBuildProperties(buildName, buildNumber, project string, imageLayers []ut
 }
 
+// filterLayersForVirtualRepository filters image layers to only include those from the default deployment repository
+// when dealing with virtual repositories. For non-virtual repositories, it returns all layers unchanged.
+func filterLayersForVirtualRepository(imageLayers []utils.ResultItem, serviceManager artifactory.ArtifactoryServicesManager, originalRepo string, repoDetails *RepositoryDetails) ([]utils.ResultItem, error) {
+	if len(imageLayers) == 0 {
+		return imageLayers, nil
+	}
+
+	// Optimization: If we already know the repo type and it's not virtual, skip the API call
+	if repoDetails != nil && repoDetails.repoType != "" && repoDetails.repoType != "virtual" {
+		log.Info("OPTIMIZATION: Repository", originalRepo, "is not virtual (type:", repoDetails.repoType+"), skipping API call and proceeding with all layers")
+		return imageLayers, nil
+	}
+
+	// For backwards compatibility or when repoDetails is not available, fall back to API call
+	if repoDetails == nil || repoDetails.repoType == "" {
+		log.Debug("Repository type not cached, making API call to determine repository configuration")
+		// Check if the original repository is a virtual repository by getting its configuration
+		repoConfig, err := getRepositoryConfiguration(originalRepo, serviceManager)
+		if err != nil {
+			log.Debug("Failed to get repository configuration for", originalRepo, ":", err.Error())
+			return imageLayers, err
+		}
+
+		// If it's not a virtual repository, return all layers unchanged
+		if repoConfig == nil || repoConfig.Rclass != "virtual" {
+			log.Debug("Repository", originalRepo, "is not virtual, proceeding with all layers")
+			return imageLayers, nil
+		}
+
+		// If it's a virtual repository but has no default deployment repo, return all layers
+		if repoConfig.DefaultDeploymentRepo == "" {
+			log.Debug("Virtual repository", originalRepo, "has no default deployment repository, proceeding with all layers")
+			return imageLayers, nil
+		}
+
+		// Filter layers to only include those from the default deployment repository
+		var filteredLayers []utils.ResultItem
+		for _, layer := range imageLayers {
+			if layer.Repo == repoConfig.DefaultDeploymentRepo {
+				filteredLayers = append(filteredLayers, layer)
+			}
+		}
+
+		if len(filteredLayers) > 0 {
+			log.Info("Filtered", len(imageLayers), "layers to", len(filteredLayers), "layers from default deployment repository:", repoConfig.DefaultDeploymentRepo)
+		} else {
+			log.Warn(fmt.Sprintf(`No layers found in default deployment repository '%s' for virtual repository '%s'.
+This may indicate that image layers exist in other repositories but not in the default deployment repository.
+Properties will not be set to maintain consistency with virtual repository configuration.
+To fix this, consider pushing the image directly to the virtual repository to ensure it lands in the default deployment repository.`, repoConfig.DefaultDeploymentRepo, originalRepo))
+			return []utils.ResultItem{}, nil
+		}
```

Comment:
```suggestion
		if len(filteredLayers) == 0 {
		    log.Warn(fmt.Sprintf(`No layers found in default deployment repository '%s' for virtual repository '%s'.
This may indicate that image layers exist in other repositories but not in the default deployment repository.
Properties will not be set to maintain consistency with virtual repository configuration.
To fix this, consider pushing the image directly to the virtual repository to ensure it lands in the default deployment repository.`, repoConfig.DefaultDeploymentRepo, originalRepo))
			return []utils.ResultItem{}, nil
		} 
		log.Info("Filtered", len(imageLayers), "layers to", len(filteredLayers), "layers from default deployment repository:", repoConfig.DefaultDeploymentRepo)
```